### PR TITLE
TEST-#3869: add build-docs check in CI

### LINKED ISSUE #3869: Build documentation in CI
The documentation contains many links, and their number is growing rapidly. It is no longer convenient to check their operability manually.
It is necessary to automate the procedure for checking.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

1. For now, after building docs, we can simply look at the current warnings. After they are fixed (in a separate pull request), we should add `-W --keep-going` so that any warning leads to an error.

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #3869 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/getting_started/troubleshooting.rst [hunk 1]
```diff
@@ -209,9 +209,9 @@ This can happen when you use OmniSci engine along with ``pyarrow.gandiva``:
   cfg.IsExperimental.put(True)
   import modin.pandas as pd
   import pyarrow.gandiva as gandiva  # Error
-  CommandLine Error: Option 'enable-vfe' registered more than once!
-  LLVM ERROR: inconsistency in registered CommandLine options
-  Aborted (core dumped)
+  # CommandLine Error: Option 'enable-vfe' registered more than once!
+  # LLVM ERROR: inconsistency in registered CommandLine options
+  # Aborted (core dumped)
```
Comment (anmyachev) [inline docs/getting_started/troubleshooting.rst:214]: These lines did not allow to recognize that this is a block of python code.

File: docs/getting_started/faq.rst [hunk 2]
```diff
@@ -51,7 +51,7 @@ with dataframes that don't fit into the available memory. As a result, pandas wo
 for prototyping on a few MBs of data do not scale to tens or hundreds of GBs (depending on the size 
 of your machine). Modin supports operating on data that does not fit in memory, so that you can comfortably 
 work with hundreds of GBs without worrying about substantial slowdown or memory errors. For more information, 
-see :doc:`out-of-memory support <getting_started/out_of_core.rst>` for Modin.
+see :doc:`out-of-memory support </getting_started/out_of_core>` for Modin.
```
Comment (anmyachev) [inline docs/getting_started/faq.rst:54]: Invalid link

File: docs/flow/modin/experimental/pandas.rst [hunk 3]
```diff
@@ -12,5 +12,4 @@ Experimental API Reference
 .. autofunction:: read_sql
 .. autofunction:: read_csv_glob
 .. autofunction:: read_pickle_distributed
-.. autoclass:: DataFrame
-   :members: to_pickle_distributed
```
Comment (anmyachev) [inline docs/flow/modin/experimental/pandas.rst:16]: Thus, the library generated 2 targets with the same `DataFrame` name, which resulted in a warning. (`...\projects\modin\modin\pandas\base.py:docstring of modin.pandas.base.BasePandasDataset.var:: WARNING: more than one target found for cross-reference 'DataFrame': modin.experimental.pandas.DataFrame, modin.pandas.dataframe.DataFrame`)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -34,6 +34,20 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7.x"
+          architecture: "x64"
+      - run: pip install -r docs/requirements-doc.txt
+      - run: cd docs && sphinx-build -b html . build
```
Comment (dchigarev) [inline .github/workflows/ci.yml:49]:
```suggestion
      - run: cd docs && sphinx-build -T -E -b html . build
```
1. `-T` should [increase verbosity](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-T) if something goes wrong.
2. `-E` [ensures](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-E) that we're not using any cache.

**Review by dchigarev — CHANGES_REQUESTED** [review]

**Review by dchigarev — APPROVED** [review]
LGTM!

**Comment by anmyachev** [comment]
@modin-project/modin-core let's merge if no other comments :)

**Review by devin-petersohn — APPROVED** [review]

**Pull request merged**